sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EncodingExtensions.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Text;
+
+namespace Azure.Messaging.EventHubs.Primitives
+{
+    internal static class EncodingExtensions
```

Comment:
These methods are actually available on NETStandard 2.1 and NetCoreApp. Should we ifdef here? If so any advises what types of conditions we would normally use?

Example

https://docs.microsoft.com/en-us/dotnet/api/system.text.encoding.getbytes?view=net-6.0#system-text-encoding-getbytes(system-readonlyspan((system-char))-system-span((system-byte)))

Reply:
Well the benefit of ifdef would be that instead of using the local implementation the code path would be using the framework built-in versions and hence automatically benefit from any optimizations being done there

Reply:
The question that I'm considering is whether those potential optimizations are likely to be impactful enough that it warrants the lack of testability that we'd have and, to a lesser extend, warrants future maintainers (including future me) having to reason about what's being called.   If we continue to accept that the big win is that you got us to non-allocating, would saving a couple of ns make an impact?   On the other hand, wrapping the entire class in an `#ifdef` with a comment explaining things isn't a ton of cognitive burden or complexity.

I remain pretty neutral, but I'd slant towards leaving it without the `#ifdef`, potentially with a comment to reconsider if we start to explicitly support `netstandard2.1` as a target.